js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
+        <div
+          className={ classes.join(' ') }
+          style={ { top, left } }
+          onKeyDown={ this.handleKeyDown }
+        >
+          <ParityBackground muiTheme={ muiTheme } className={ styles.parityBackground } />
+          <div className={ styles.inputContainer }>
+            <label className={ styles.label } htmlFor={ id }>
+              { label }
+            </label>
+            <input
+              id={ id }
+              className={ styles.input }
+              placeholder={ hint }
+
+              onBlur={ this.handleInputBlur }
+              onFocus={ this.handleInputFocus }
+              onChange={ this.handleChange }
+
+              ref={ this.setInputRef }
+            />
+
+            <div className={ styles.underline }>
+              <TextFieldUnderline
+                muiTheme={ muiTheme }
+                focus={ inputFocused }
+                style={ {
+                  borderBottom: 'solid 3px'
+                } }
+                focusStyle={ {
+                  borderBottom: 'solid 3px'
+                } }
+              />
+            </div>
+
+            { this.renderCurrentInput() }
+            { this.renderAccounts() }
+            { this.renderCloseIcon() }
+          </div>
+        </div>
+      </Portal>
     );
   }
 
-  renderItem = (entry) => {
-    const { address, name } = entry;
+  renderCurrentInput () {
+    if (!this.props.allowInput) {
+      return null;
+    }
 
-    const _balance = this.getBalance(address);
-    const balance = _balance ? _balance.toNumber() : _balance;
+    const { inputValue } = this.state;
 
-    if (!this.items[address] || this.items[address].balance !== balance) {
-      this.items[address] = {
-        text: name && name.toUpperCase() || address,
-        value: this.renderMenuItem(address),
-        address, balance
-      };
+    if (inputValue.length === 0 || !/^(0x)?[a-f0-9]*$/i.test(inputValue)) {
+      return null;
     }
 
-    return this.items[address];
+    return (
+      <div
+
+      >
+        { this.renderAccountCard({ address: inputValue }) }
+      </div>
+    );
   }
 
-  getBalance (address) {
-    const { balances = {} } = this.props;
-    const balance = balances[address];
+  renderCloseIcon () {
+    const { expanded } = this.state;
 
-    if (!balance) {
+    if (!expanded) {
       return null;
     }
 
-    const ethToken = balance.tokens.find((tok) => tok.token && tok.token.tag && tok.token.tag.toLowerCase() === 'eth');
+    return (
+      <div className={ styles.closeIcon } onClick={ this.handleClose }>
+        <CloseIcon style={ { width: 48, height: 48 } } />
+      </div>
+    );
+  }
 
-    if (!ethToken) {
-      return null;
+  renderAccounts () {
+    const { values } = this.state;
+
+    if (values.length === 0) {
+      return (
+        <div className={ styles.categories }>
+          <div className={ styles.empty }>
+            No account matches this query...
```

Comment:
New components should really use `FormattedMessage` so we don't need to come back to it.